server.py

```diff
@@ -1,20 +1,24 @@
 # -*- coding: utf-8 -*-
-from __future__ import unicode_literals
 import socket
 import sys
+import os
+import mimetypes
 
-
-ADDR = ('127.0.0.1', 8001)
-CRLF = ('\r\n')
+ROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'webroot')
+ROOT = ROOT.encode('utf-8')
+ADDR = (b'127.0.0.1', 8004)
+CRLF = (b'\r\n')
 PROTOCOL = b'HTTP/1.1'
 # Using this as a dummy var
-foo_date = "Sun, 21 Jul 2001 23:32:15 GTM"
-reqtypes = set(["POST", "GET", "PUT", "HEAD", "DELETE", "OPTIONS", "TRACE"])
+foo_date = b"Sun, 21 Jul 2001 23:32:15 GTM"
+reqtypes = set([b"POST", b"GET", b"PUT", b"HEAD", b"DELETE", b"OPTIONS",
+    b"TRACE"])
+
 
 RESPONSE = CRLF.join([
     b'HTTP/1.1 {response_code} {response_reason}',
-    b'Content-Type: text/html; charset=UTF-8',
-    b'Date: {date}', CRLF])
+    b'Content-Type: {ctype}; charset=UTF-8',
```

Comment:
Remember, the `charset` extension to the `Content-Type` header is really only good if the content being delivered is text.  If we are dealing with a binary file, like an image, this information would be incorrect.